test/local.mk

```
@@ -0,0 +1,104 @@
+crossplane_sentinel = $(kind_dir)/crossplane_sentinel
+registry_sentinel = $(kind_dir)/registry_sentinel
+
+setup_envtest_bin = $(kind_dir)/setup-envtest
+
+# Prepare binary
+# We need to set the Go arch since the binary is meant for the user's OS.
+$(setup_envtest_bin): export GOOS = $(shell go env GOOS)
+$(setup_envtest_bin): export GOARCH = $(shell go env GOARCH)
+$(setup_envtest_bin):
+	@mkdir -p $(kind_dir)
+	cd test && go build -o $@ sigs.k8s.io/controller-runtime/tools/setup-envtest
+	$@ $(ENVTEST_ADDITIONAL_FLAGS) use '$(ENVTEST_K8S_VERSION)!'
+	chmod -R +w $(kind_dir)/k8s
+
+.PHONY: local-install
+local-install: export KUBECONFIG = $(KIND_KUBECONFIG)
+# for ControllerConfig:
+local-install: export INTERNAL_PACKAGE_IMG = registry.registry-system.svc.cluster.local:5000/$(PROJECT_OWNER)/$(PROJECT_NAME)/package:$(IMG_TAG)
+# for package-push:
+local-install: PACKAGE_IMG = localhost:5000/$(PROJECT_OWNER)/$(PROJECT_NAME)/package:$(IMG_TAG)
+local-install: kind-load-image crossplane-setup registry-setup $(kind_dir)/.credentials.yaml package-push  ## Install Operator in local cluster
+	yq e '.spec.metadata.annotations."local.dev/installed"="$(shell date)"' test/controllerconfig-exoscale.yaml | kubectl apply -f -
+	yq e '.spec.package=strenv(INTERNAL_PACKAGE_IMG)' test/provider-exoscale.yaml | kubectl apply -f -
+	kubectl wait --for condition=Healthy provider.pkg.crossplane.io/provider-exoscale --timeout 60s
+	kubectl -n crossplane-system wait --for condition=Ready $$(kubectl -n crossplane-system get pods -o name -l pkg.crossplane.io/provider=provider-exoscale) --timeout 60s
+	kubectl apply -n crossplane-system -f $(kind_dir)/.credentials.yaml
```

Apply the credentials earlier
```suggestion
	kubectl apply -n crossplane-system -f $(kind_dir)/.credentials.yaml
	yq e '.spec.metadata.annotations."local.dev/installed"="$(shell date)"' test/controllerconfig-exoscale.yaml | kubectl apply -f -
	yq e '.spec.package=strenv(INTERNAL_PACKAGE_IMG)' test/provider-exoscale.yaml | kubectl apply -f -
	kubectl wait --for condition=Healthy provider.pkg.crossplane.io/provider-exoscale --timeout 60s
	kubectl -n crossplane-system wait --for condition=Ready $$(kubectl -n crossplane-system get pods -o name -l pkg.crossplane.io/provider=provider-exoscale) --timeout 60s
```